src/telemetryReporter.ts

```diff
@@ -18,9 +20,16 @@ export default class TelemetryReporter extends vscode.Disposable {
     private static TELEMETRY_CONFIG_ID = 'telemetry';
     private static TELEMETRY_CONFIG_ENABLED_ID = 'enableTelemetry';
 
+    private logFilePath: string;
+    private logStream: fs.WriteStream;
+
     constructor(private extensionId: string, private extensionVersion: string, key: string) {
         super(() => this.toDispose.forEach((d) => d && d.dispose()))
-
+        this.logFilePath = process.env['VSCODE_LOGS'] || '';
+        if (this.logFilePath) {
+            this.logFilePath = path.join(this.logFilePath, `${extensionId}.txt`);
```

Comment:
`extensionId` shouldn't be the empty string but I don't think we enforce this strictly. We shouldn't attempt to create a log file `.txt`.

Reply:
Done